src/components/SafeAddressSelector/SafeAddressSelector.tsx

```diff
@@ -0,0 +1,79 @@
+import { ReactElement, ChangeEvent } from 'react'
+import styled from 'styled-components'
+import { EthHashInfo } from '@gnosis.pm/safe-react-components'
+import FormControl from '@material-ui/core/FormControl'
+import MenuItem from '@material-ui/core/MenuItem'
+import Select from '@material-ui/core/Select'
+import ExpandMore from '@material-ui/icons/ExpandMore'
+
+import { AddressBookEntry } from 'src/logic/addressBook/model/addressBook'
+import { black300 } from 'src/theme/variables'
+import { getChainById } from 'src/config'
+
+type SafeAddressSelectorProps = {
+  safes: AddressBookEntry[]
+  value: string
+  onChange: (event: ChangeEvent<{ value: string }>) => void
+  shouldShowShortName?: boolean
+  dataTestid?: string
+}
+
+const SafeAddressSelector = ({
```

Comment:
Generic Selector Component that allows you to select a safe from a list of safes:

![Captura de pantalla 2022-05-04 a las 14 46 44](https://user-images.githubusercontent.com/26763673/166684055-b7bd206d-85a0-43f2-a67a-e0b208ed8df6.png)
![Captura de pantalla 2022-05-04 a las 14 46 51](https://user-images.githubusercontent.com/26763673/166684059-accc8db3-6bff-43f5-aada-fd9850795dbd.png)